flowplayer.hlsjs.js

```diff
@@ -602,7 +602,20 @@
                                             maxLevel = hls.loadLevel;
                                         }
                                         break;
-
+                                    case "FRAG_PARSING_METADATA":
+                                        data.samples.forEach(function(sample) {
+                                          var metadataHandler = function() {
+                                            if (videoTag.currentTime < sample.dts) return;
+                                            videoTag.removeEventListener('timeupdate', metadataHandler);
+                                            var raw = new TextDecoder('utf-8').decode(sample.data);
+                                            player.trigger('metadata', [player, {
+                                              key: raw.substr(10, 4),
+                                              data: raw.substr(21)
+                                            }]);
+                                          };
+                                          videoTag.addEventListener('timeupdate', metadataHandler);
```

Comment:
ok - just curious; removing a listener which potentially doesn't exist is not a problem?
Strictly speaking `metadataHandler` is out of scope, can be avoided by declaring the variable, and then assign the function; would that break the code?